### DISCUSSION TITLE
What is the reasoning for ExecStartPre=/usr/bin/sleep 15 in systemd unit?

### DISCUSSION THREAD

**Discussion by mailinglists35** [abraunegg/onedrive#2983] [Q&A] [answered]
I'm coincidentally having the same DefaultTimeoutStartSec value customized to system.conf and obviously in default service provided, the service times out starting.

Before I commend or reduce the value 15 to lower, I would like to understand the impact.

Is there a reason why that delay in execstartpre, or any delay at all?

Thanks

**Comment by abraunegg** [comment, marked as answer]
@mailinglists35 
> Is there a reason why that delay in execstartpre, or any delay at all?


As per the changelog / release notes - https://github.com/abraunegg/onedrive/releases/tag/v2.5.2 

* Added 15 second sleep to systemd services to allow d-bus daemon to start and be available if present

The relevant discussion can be found here: https://github.com/abraunegg/onedrive/discussions/2762

**Reply by mailinglists35** [reply]
thank you. it seems than in my case where I only run it headless it does not affect the operation at all if I comment out the startup delay. still warns in logs about dbus unavailable but that's all
